internal/Picker/Picker.js

```diff
@@ -87,610 +86,239 @@ const allowedClassNames = ['picker', 'valueWrapper', 'joined', 'horizontal', 've
-		 *
-		 * @type {Number}
-		 * @default 0
-		 * @public
-		 */
-		value: PropTypes.number,
-
-		/**
-		 * Choose a specific size for your picker. `'small'`, `'medium'`, `'large'`, or set to `null` to
-		 * assume auto-sizing. `'small'` is good for numeric pickers, `'medium'` for single or short
-		 * word pickers, `'large'` for maximum-sized pickers.
-		 *
-		 * You may also supply a number. This number will determine the minimum size of the Picker.
-		 * Setting a number to less than the number of characters in your longest value may produce
-		 * unexpected results.
-		 *
-		 * @type {('small'|'medium'|'large'|Number)}
-		 * @public
-		 */
-		width: PropTypes.oneOfType([
-			PropTypes.oneOf([null, 'small', 'medium', 'large']),
-			PropTypes.number
-		]),
-
-		/**
-		 * Should the picker stop incrementing when the picker reaches the last element? Set `wrap`
-		 * to `true` to allow the picker to continue from the opposite end of the list of options.
-		 *
-		 * @type {Boolean}
-		 * @public
-		 */
-		wrap: PropTypes.bool
-	};
-
-	static defaultProps = {
-		accessibilityHint: '',
-		changedBy: 'enter',
-		orientation: 'horizontal',
-		spotlightDisabled: false,
-		step: 1,
-		type: 'string',
-		value: 0
-	};
-
-	constructor (props) {
-		super(props);
-
-		this.state = {
-			// Set to `true` onFocus and `false` onBlur to prevent setting aria-valuetext (which
-			// will notify the user) when the component does not have focus
-			active: false,
-			pressed: 0
-		};
-
-		this.containerRef = createRef();
-
-		// Pressed state for this.handleUp
-		this.pickerButtonPressed = 0;
-	}
-
-	componentDidMount () {
-		if (this.props.joined) {
-			this.containerRef.current.addEventListener('wheel', this.handleWheel);
-		}
-		if (platform.webos) {
-			this.containerRef.current.addEventListener('webOSVoice', this.handleVoice);
-		}
-	}
-
-	componentDidUpdate (prevProps) {
-		if (this.props.joined && !prevProps.joined) {
-			this.containerRef.current.addEventListener('wheel', this.handleWheel);
-		} else if (prevProps.joined && !this.props.joined) {
-			this.containerRef.current.removeEventListener('wheel', this.handleWheel);
-		}
-	}
+const PickerBase = (props) => {
+	// Set to `true` onFocus and `false` onBlur to prevent setting aria-valuetext (which
+	// will notify the user) when the component does not have focus
+	const [active, setActive] = useState(false);
+	const [pressed, setPressed] = useState(0);
+	const containerRef = useRef();
+	const pickerButtonPressed = useRef(0);
+	const reverseTransition = useRef(false);
+
+	const {
+		'aria-valuetext': ariaValueText,
+		changedBy,
+		children,
+		css: customCss,
+		disabled,
+		id,
+		index,
+		joined,
+		max,
+		min,
+		onChange,
+		onSpotlightDisappear,
+		orientation,
+		reverse,
+		spotlightDisabled,
+		step,
+		value,
+		width,
+		wrap,
+		...rest
+	} = props;
+	const css = usePublicClassNames({componentCss, customCss, publicClassNames: true});
+	const decrementIcon = selectDecIcon(props);
```

Comment:
I think the order of this definition is not necessarily alphabetical. Can we keep the previous order?

Reply:
Alex is on holiday, so I pushed the changes, but I am not sure I understood right what to change with this comment. Please have a look.